src/Facets/GasZipFacet.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import { ILiFi } from "../Interfaces/ILiFi.sol";
+import { LibSwap } from "../Libraries/LibSwap.sol";
+import { LibAsset } from "../Libraries/LibAsset.sol";
+import { ReentrancyGuard } from "../Helpers/ReentrancyGuard.sol";
+import { SwapperV2 } from "../Helpers/SwapperV2.sol";
+import { Validatable } from "../Helpers/Validatable.sol";
+import { SafeTransferLib } from "solady/utils/SafeTransferLib.sol";
+import { ERC20 } from "solady/tokens/ERC20.sol";
+import { NativeAssetTransferFailed } from "lifi/Errors/GenericErrors.sol";
+
+interface IGasZip {
+    function deposit(
+        uint256 destinationChains,
+        address recipient
+    ) external payable;
+}
+
+/// @title GasZipFacet
+/// @author LI.FI (https://li.fi)
+/// @notice Provides functionality to swap ERC20 tokens to native and deposit them to the  gas.zip protocol (https://www.gas.zip/)
+/// @custom:version 1.0.0
+contract GasZipFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable {
+    using SafeTransferLib for address;
+
+    /// @dev GasZip-specific bridge data
+    /// @param destinationChains a value that represents a list of chains to which gas should be distributed (see https://dev.gas.zip/gas/code-examples/deposit for more details)
+    /// @param gasZipSwapData (only required for ERC20 tokens): the swapData that swaps from ERC20 to native before depositing to gas.zip
+    struct GasZipData {
+        uint256 destinationChains;
+        LibSwap.SwapData gasZipSwapData;
+    }
+
+    /// State ///
+    IGasZip public immutable gasZipRouter;
+
+    /// Constructor ///
+    constructor(address _gasZipRouter) {
+        gasZipRouter = IGasZip(_gasZipRouter);
+    }
+
+    /// @notice Bridges tokens using the gas.zip protocol
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _gasZipData GasZip-specific bridge data
+    function startBridgeTokensViaGasZip(
+        ILiFi.BridgeData memory _bridgeData,
+        GasZipData calldata _gasZipData
```

Comment:
I would not pass `GasZipData` including `gasZipSwapData` to the bridge functions as we are not using that at all.

Reply:
mhm....not sure if I can agree.

There are two cases: 
1) gaszip native(here we dont have calldata in `_gasZipData`, it's just an empty parameter, e.g. `0x`)
2) gaszip ERC20 (here we have calldata in `_gasZipData`)

Both of these cases can either be just a bridge tx, or a with a swapStep beforehand.

For me it makes sense the way it is.
Could you please reconsider and elaborate (or call me)? 

What I did notice though is that we dont need the destinationChains parameter in the `gasZipData` if we will never use it as a combined value for several chains. So I removed this and use `bridgeData.destinationChainId` instead which will save us some gas

Reply:
Hye, "not using" was perhaps the wrong term. We don't need it. `swapAndStartBridgeTokensViaGasZip` should be used to swap tokens to native and has all the handling to pass along the swap result.

I would recommend to use `startBridgeTokensViaGasZip` only for native tokens and `swapAndStartBridgeTokensViaGasZip` for all other ERC20s. 

Please note that the chainIds gas.zip is using differ from the normal chainIds we use. We either have to maintain a mapping or pass their id:
https://dev.gas.zip/gas/chain-support/outbound

Reply:
Updated as discussed yesterday. Please re-review and approve @maxklenk